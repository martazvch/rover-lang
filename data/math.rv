struct Vec2 {
    x: int,
    y: int,

    fn len(self) -> int {
        self.x + self.y
    }
}

fn introduce() {
    print "This is a mathematical module"
}

fn getVecFromPos(x: int, y: int) -> Vec2 {
    Vec2{ x, y }
}

fn getInt() -> int { 12 }

fn getIntDef(a = 5) -> int { 12 }


