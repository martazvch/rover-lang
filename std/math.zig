pub fn cos(x: f64) f64 {
    return @cos(x);
}

pub fn sin(x: f64) f64 {
    return @sin(x);
}
