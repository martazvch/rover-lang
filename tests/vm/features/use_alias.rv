use .data.math as m

fn main() {
    m.introduce()
    var vec = m.getVecFromPos(1, 2)
    print vec.x
    print vec.y
    vec = m.getVecFromPos(8, 9)
    print vec.x
    print vec.y

    fn trash(a: int) -> bool {
        return a < 10
    }
    print trash(vec.y)

    vec = m.getVecFromPos(3, 7)
    print vec.x
    print vec.y

    var please = m.introduce
    please()

    var please2 = m.getVecFromPos
    var vec2 = please2(6, 89)

    print vec2.x
    print vec2.y
}
