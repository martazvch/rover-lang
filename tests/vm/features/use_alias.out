"This is a mathematical module"
1
2
8
9
true
3
7
"This is a mathematical module"
6
89

